health-services/project-factory/src/server/utils/onGoingCampaignUpdateUtils.ts

```
@@ -0,0 +1,806 @@
+
+async function getBoundariesFromCampaignSearchResponse(request: any, campaignDetails: any) {
+  let parentCampaignBoundaries: any[] = [];
+  if (campaignDetails?.parentId) {
+    const parentCampaignObject = await getParentCampaignObject(request, campaignDetails?.parentId);
+    parentCampaignBoundaries = parentCampaignObject?.boundaries;
+  }
+  return getBoundariesArray(parentCampaignBoundaries, campaignDetails?.boundaries)
+}
+
+async function fetchProjectsWithParentRootProjectId(request: any) {
+  const { projectId, tenantId } = request?.body?.parentCampaign;
```

Biome

[error] Unsafe usage of optional chaining.

If it short-circuits with 'undefined' the evaluation will throw TypeError here:


(lint/correctness/noUnsafeOptionalChaining)